x-pack/platform/plugins/private/upgrade_assistant/server/lib/es_deprecations_status/index.ts

```diff
@@ -56,14 +56,22 @@ export async function getESUpgradeStatus(
       return status !== 'green';
     }) as EnrichedDeprecationInfo[];
 
-    return [...enrichedHealthIndicators, ...toggledMigrationsDeprecations];
+    return {
+      enrichedHealthIndicators,
+      migrationsDeprecations: toggledMigrationsDeprecations,
+    };
   };
+  const { enrichedHealthIndicators, migrationsDeprecations } = await getCombinedDeprecations();
 
-  const combinedDeprecations = await getCombinedDeprecations();
-  const criticalWarnings = combinedDeprecations.filter(({ isCritical }) => isCritical === true);
-
-  return {
-    totalCriticalDeprecations: criticalWarnings.length,
-    deprecations: combinedDeprecations,
+  const result = {
+    totalCriticalDeprecations: migrationsDeprecations.filter(
```

Comment:
Keeping these indicators separate rather than having to filter for health indicators again in the `status` handler.